### PR TITLE
[wni] bump dependencies to support OSDK v0.18

### PR DESCRIPTION
**Pull request by PratikMahajan**
we need to bump the dependencies in WNI in order to upgrade the
operator-sdk to v0.18 in windows-machine-config-operator repo.

there are breaking changes in client-go and openshift client-go that is
mandating this bump for dependencies.

updated client-go function calls, which have signature changes related to
passing contexts and options.

### CODE REVIEW

**Comment by VaishnaviHire** [comment]
LGTM

File: tools/windows-node-installer/test/e2e/clustersetup.go [hunk 1]
```diff
@@ -215,27 +216,27 @@ func deleteResources(kubeconfig string) error {
 	core, _ := coreclient.NewForConfig(cfg)
 
 	// delete role binding
-	if err := rbacConfig.RoleBindings(projectName).Delete(resourceName, &metav1.DeleteOptions{}); err != nil {
+	if err := rbacConfig.RoleBindings(projectName).Delete(context.TODO(), resourceName, metav1.DeleteOptions{}); err != nil {
```
Comment (mansikulkarni96) [inline tools/windows-node-installer/test/e2e/clustersetup.go:219]: is there a reason for not passing a reference of &metav1.DeleteOptions{} to Delete anymore?

File: tools/windows-node-installer/pkg/client/openshift.go [hunk 2]
```diff
@@ -68,7 +69,7 @@ func (o *OpenShift) GetCloudProvider() (*v1.PlatformStatus, error) {
 
 // getInfrastructure returns the information of current Infrastructure referred by the OpenShift client or an error.
 func (o *OpenShift) getInfrastructure() (*v1.Infrastructure, error) {
-	infra, err := o.Client.ConfigV1().Infrastructures().Get("cluster", metav1.GetOptions{})
+	infra, err := o.Client.ConfigV1().Infrastructures().Get(context.TODO(), "cluster", metav1.GetOptions{})
```
Comment (sebsoto) [inline tools/windows-node-installer/pkg/client/openshift.go:72]: Is it best practice to use `context.TODO()`?

File: tools/windows-node-installer/go.mod [hunk 3]
```diff
@@ -3,11 +3,11 @@ module github.com/openshift/windows-machine-config-bootstrapper/tools/windows-no
 go 1.13
 
 replace (
-	github.com/openshift/api => github.com/openshift/api v0.0.0-20200205145930-e9d93e317dd1 // OpenShift 4.3
-	github.com/openshift/client-go => github.com/openshift/client-go v0.0.0-20191125132246-f6563a70e19a // OpenShift 4.3
-	k8s.io/api => k8s.io/api v0.16.7
-	k8s.io/apimachinery => k8s.io/apimachinery v0.16.7
-	k8s.io/client-go => k8s.io/client-go v0.16.7
+	github.com/openshift/api => github.com/openshift/api v0.0.0-20200422081840-fdd1b0c14c88
+	github.com/openshift/client-go => github.com/openshift/client-go v0.0.0-20200422192633-6f6c07fc2a70
```
Comment (aravindhp) [inline tools/windows-node-installer/go.mod:7]: Please add a comment saying which OpenShift version this corresponds to.

File: tools/windows-node-installer/pkg/client/openshift.go (hunk 2 above)
Comment (aravindhp) [reply tools/windows-node-installer/pkg/client/openshift.go:72]: The [context](https://golang.org/pkg/context/#pkg-overview) pkg overview gives guidance on this.

**Comment by aravindhp** [comment]
@PratikMahajan please wait until #206 merges before you remove the hold.

**Comment by PratikMahajan** [comment]
@ravisantoshgudimetla 
> Mostly looks good. I am curious if you've run `go mod tidy` which brings in all the latest changes. In particular, we should try to get k8s.io/apimachinery which corresponds to the one that WMCO's controller -runtime brings in when bumped to v0.18

Yes, I've ran `go mod tidy`

File: tools/windows-node-installer/go.mod (hunk 3 above)
Comment (PratikMahajan) [reply tools/windows-node-installer/go.mod]: added

**Pull request merged**